packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview OSA Script to interact with Xcode. Functionality includes
+ * checking if a given project is open in Xcode, starting a debug session for
+ * a given project, and stopping a debug session for a given project.
+ */
+
+'use strict';
+
+function run(args_array = []) {
+  let args;
+  try {
+    args = new CommandArguments(args_array);
+  } catch (e) {
+    return new RunJsonResponse(false, `Failed to parse arguments: ${e}`);
+  }
+
+  const xcodeResult = getXcode(args);
+  if (xcodeResult.error != null) {
+    return new RunJsonResponse(false, xcodeResult.error).stringify();
+  }
+  const xcode = xcodeResult.result;
+
+  if (args.command === 'project-opened') {
+    const result = getWorkspace(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else if (args.command === 'debug') {
+    const result = debugApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error, result.result).stringify();
+  } else if (args.command === 'stop') {
+    const result = stopApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else {
+    return new RunJsonResponse(false, 'Unknown command').stringify();
+  }
+}
+
+/**
+ * Parsed and validated arguments passed from the command line.
+ */
+class CommandArguments {
+  /**
+   *
+   * @param {!Array<string>} args List of arguments passed from the command line.
+   */
+  constructor(args) {
+    this.command = this.validatedCommand(args[0]);
+
+    const parsedArguments = this.parseArguments(args);
+
+    this.xcodePath = this.validatedStringArgument('--xcode-path', parsedArguments['--xcode-path']);
+    this.projectPath = this.validatedStringArgument('--project-path', parsedArguments['--project-path']);
+    this.workspacePath = this.validatedStringArgument('--workspace-path', parsedArguments['--workspace-path']);
+    this.targetDestinationId = this.validatedStringArgument('--device-id', parsedArguments['--device-id']);
+    this.targetSchemeName = this.validatedStringArgument('--scheme', parsedArguments['--scheme']);
+    this.skipBuilding = this.validatedBoolArgument('--skip-building', parsedArguments['--skip-building']);
+    this.launchArguments = this.validatedJsonArgument('--launch-args', parsedArguments['--launch-args']);
+    this.closeWindowOnStop = this.validatedBoolArgument('--close-window', parsedArguments['--close-window']);
+    this.promptToSaveBeforeClose = this.validatedBoolArgument('--prompt-to-save', parsedArguments['--prompt-to-save']);
+    this.verbose = this.validatedBoolArgument('--verbose', parsedArguments['--verbose']);
```

Comment:
are these existing args that we already support?

Reply:
I'm not sure what you mean. These are arguments used throughout the script to do various things, so it's all new

Reply:
im wondering if there's any place we document these APIs.

Reply:
to clarify, this is an internal script only intended to be invoked by the tool. if any of these are ambiguous, we can clarify with a comment, but these seem pretty self-evident to me?